inc/zoo/swar/math.h

```diff
@@ -0,0 +1,69 @@
+#pragma once
+#include "SWAR.h"
+
+namespace zoo::math {
+
+template <typename IntegerType = size_t>
+constexpr static
+std::enable_if_t<std::is_integral_v<IntegerType>, bool>
+isPowerOfTwo(IntegerType x) noexcept {
+    return x && (x & (x - 1)) == 0;
+}
+
+template <typename IntegerType = size_t, IntegerType X>
+constexpr static
+std::enable_if_t<std::is_integral_v<IntegerType>, bool>
+isPowerOfTwo() noexcept {
+    return isPowerOfTwo(X);
+}
+static_assert(isPowerOfTwo<int, 4>());
+
+
+template <size_t N, typename IntegerType = size_t>
+constexpr static
+std::enable_if_t<
+    std::is_integral_v<IntegerType> &&
+    isPowerOfTwo<size_t, N>(), size_t>
+moduloPowerOfTwo(IntegerType x) noexcept {
+    return x & (N - 1);
+}
+
+static_assert(moduloPowerOfTwo<4>(0) == 0);
+static_assert(moduloPowerOfTwo<8>(9) == 1);
+static_assert(moduloPowerOfTwo<4096>(4097) == 1);
```

Comment:
I know we have some pollution like this in our headers.  It is pollution we haven't bother to clean.  Please move these legitimate compile time tests to their "translation unit" out of the header.

Reply:
Good callout, thanks.